### PR TITLE
check: scrub deleted document version from core

### PR DESCRIPTION
**Pull request by spolu**
Instead of running it manually from time to time, run it every 8h automatically.

### CODE REVIEW

**Comment by spolu** [comment]
This replace: https://github.com/dust-tt/dust/blob/main/front/migrations/20231110_scrub_deleted_documents.ts

File: front/production_checks/temporal/activities.ts [hunk 1]
```diff
@@ -30,29 +39,42 @@ async function runAllChecks(checks: Check[]) {
...
-        uuid: uuid,
-      });
-      await check.check(check.name, reportSuccess, reportFailure);
-      Context.current().heartbeat({
-        type: "finish",
-        name: check.name,
-        uuid: uuid,
-      });
+      const currentHour = new Date().getHours();
+      if (currentHour % check.everyHour !== 0) {
```
Comment (lasryaric) [inline front/production_checks/temporal/activities.ts:42]: Super nice!

File: front/production_checks/checks/scrub_deleted_core_document_versions.ts [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+    throw new Error("Env var DUST_DATA_SOURCES_BUCKET is not defined");
+  }
+
+  const core_sequelize = new Sequelize(CORE_DATABASE_URI as string, {
+    logging: false,
+  });
+
+  const deletedDocumentsData = await core_sequelize.query(
+    `SELECT * FROM data_sources_documents WHERE status = 'deleted'`
+  );
```
Comment (lasryaric) [inline front/production_checks/checks/scrub_deleted_core_document_versions.ts:32]: Side note: you will get back a `object[]` type if you add  `{ type: QueryTypes.SELECT }`:
```
.query(`...`, , { type: QueryTypes.SELECT })
```

File: front/production_checks/checks/scrub_deleted_core_document_versions.ts [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+  hash: string
+) {
+  // Process same version of same document only once.
+  const uid = `${data_source}-${document_id}-${hash}`;
+  if (seen.has(uid)) {
+    return false;
+  }
+
+  const moreRecentSameHash = await core_sequelize.query(
+    `SELECT id FROM data_sources_documents WHERE data_source = :data_source AND document_id = :document_id AND hash = :hash AND status != 'deleted' AND created > :deletedAt LIMIT 1`,
```
Comment (lasryaric) [inline front/production_checks/checks/scrub_deleted_core_document_versions.ts:103]: might be worth to a >= for the deletedAt time? If done programmatically, we could end up with the same.

File: front/production_checks/checks/scrub_deleted_core_document_versions.ts [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+  const hasher = createHash();
+  hasher.update(Buffer.from(document_id));
+  const documentIdHash = hasher.digest("hex");
+
+  const path = `${dataSource.project}/${dataSource.internal_id}/${documentIdHash}/${hash}`;
+
+  const storage = new Storage({ keyFilename: SERVICE_ACCOUNT });
+
+  const [files] = await storage
+    .bucket(DUST_DATA_SOURCES_BUCKET || "")
```
Comment (lasryaric) [inline front/production_checks/checks/scrub_deleted_core_document_versions.ts:147]: why `|| ""` ? Don't we want to error if `DUST_DATA_SOURCES_BUCKET` is falsy?

File: front/production_checks/checks/scrub_deleted_core_document_versions.ts [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+        hash: hash,
+      },
+    }
+  );
+
+  if (files.length === 0) {
+    return false;
+  }
+
+  await Promise.all(
```
Comment (lasryaric) [inline front/production_checks/checks/scrub_deleted_core_document_versions.ts:174]: Delete on GCS before deleting in our Database.

**Review by lasryaric — CHANGES_REQUESTED** [review]
Left a few comments.

File: front/production_checks/checks/scrub_deleted_core_document_versions.ts (hunk 3 above)
Comment (spolu) [reply front/production_checks/checks/scrub_deleted_core_document_versions.ts:103]: :+1:

File: front/production_checks/checks/scrub_deleted_core_document_versions.ts (hunk 4 above)
Comment (spolu) [reply front/production_checks/checks/scrub_deleted_core_document_versions.ts:147]: undefined to be precise

Comment (lasryaric) [reply front/production_checks/checks/scrub_deleted_core_document_versions.ts:147]: sure but if it is, do we still want to proceed with the getFiles() / Delete() call?

**Review by lasryaric — COMMENTED** [review]
One more question about `DUST_DATA_SOURCES_BUCKET` being undefined, LGTM otherwise.

**Pull request merged**